lib/starknet_explorer/block/block.ex

```diff
@@ -137,6 +188,45 @@ defmodule StarknetExplorer.Block do
     struct(__MODULE__, rpc_block)
   end
 
+  @doc """
+  This function will return the lowest continuous block starting from the highest one
+  stored so far. Note this is not the same as the lowest block. Example:
+
+  If we have stored the blocks [5, 6, 20, 21, 22], this returns `20`, not `5`.
+  We are using this for the block fetcher logic, where we want to go downwards in order.
+  The problem is a block with a lower number could be added by someone visiting a details
+  page for a block, so we need to account for that.
+  """
+  def get_lowest_block_number(network) do
+    Repo.query(
+      "SELECT  number - 1
+    FROM    blocks block
+    WHERE   NOT EXISTS
+            (
+            SELECT  NULL
+            FROM    blocks mi
+            WHERE   mi.number = block.number - 1 AND mi.network = $1
+            ) AND block.network = $1
+    ORDER BY number DESC
+    LIMIT 1",
+      [network]
+    )
```

Comment:
This looks like a fairly standard query but we might want to make sure it works on sqlite as well if we want to keep supporting that

Reply:
Yeah, this was fixed here [adb876d](https://github.com/lambdaclass/madara_explorer/pull/273/commits/adb876dd9003ece2cc1ef969abaa4f15a7887aac)